files/common/lib/systemd/system/delphix-migration.service

```diff
@@ -37,11 +37,22 @@
 # local-fs.target, it is better to be more explicit in case we decided to
 # remove the latter.
 #
+# The OS network migration code, even though it doesn't list the
+# netplan file generated by cloud-init as a hard dependency, it can
+# experience a race as the one pointed in DLPX-67281, where the
+# cloud-init-local service runs around the same time and generates
+# the default netplan file while we write our custom one on-disk
+# through the netconf-migrator. This is bad because we end up with two
+# netplan files that can have conflicting information for networkd.
+# To avoid this scenario, we make sure that this service runs AFTER
+# cloud-init-local.
+#
 [Unit]
 Description=Delphix OS Migration Service
 PartOf=delphix.target
 DefaultDependencies=no
 After=systemd-udev-settle.service
+After=cloud-init-local.service
```

Comment:
Out of curiosity, do we have to worry about running before any other cloud-init services? In other words, does this also need a `Before=cloud-init.service`?

Reply:
That's a great question. In term of anything that we manage by netplan there should be nothing to worry about. Outside of that though, one could bring up the question of `resolv.conf`. Looking at the logs in our machines and the source code of cloud-init it is not obvious to me which service handles the creation of that file by default. If it is indeed `cloud-init` and not `cloud-init-local` then we should probably do what you suggest. I can open an issue about this so it is tracked somewhere as this is a backport and I wouldn't like to introduce any differences between this and its original commit on `master`.

Reply:
Issued https://jira.delphix.com/browse/DLPX-67388